core/util_test.go

```diff
@@ -271,52 +271,79 @@ func (d *DrandTestScenario) RunDKG() *key.Group {
 
 	leaderNode := d.nodes[0]
 	controlClient, err := net.NewControlClient(leaderNode.drand.opts.controlPort)
-	require.NoError(d.t, err)
+	required := require.New(d.t)
+	required.NoError(err)
 
 	d.t.Log("[RunDKG] Start: Leader = ", leaderNode.GetAddr())
 
 	// the leaderNode will return the group over this channel
+	errDetector := make(chan error)
+	defer close(errDetector)
 	var wg sync.WaitGroup
 	wg.Add(d.n)
 
 	// first run the leader and then run the other nodes
-	go func() {
+	runLeaderNode := func() {
+		defer wg.Done()
 		d.t.Log("[RunDKG] Leader (", leaderNode.GetAddr(), ") init")
 
 		// TODO: Control Client needs every single parameter, not a protobuf type. This means that it will be difficult to extend
 		groupPacket, err := controlClient.InitDKGLeader(
 			d.n, d.thr, d.period, d.catchupPeriod, testDkgTimeout, nil, secret, testBeaconOffset, d.scheme.ID, d.beaconID)
-		require.NoError(d.t, err)
+		if err != nil {
+			errDetector <- err
+			return
+		}
 
 		d.t.Log("[RunDKG] Leader obtain group")
 		group, err := key.GroupFromProto(groupPacket)
-		require.NoError(d.t, err)
+		if err != nil {
+			errDetector <- err
+			return
+		}
 
 		d.t.Logf("[RunDKG] Leader    Finished. GroupHash %x", group.Hash())
 
 		// We need to make sure the daemon is running before continuing
 		d.waitRunning(controlClient, leaderNode)
-		wg.Done()
-	}()
+	}
+
+	go runLeaderNode()
 
 	// all other nodes will send their PK to the leader that will create the group
 	for _, node := range d.nodes[1:] {
 		go func(n *MockNode) {
+			defer wg.Done()
 			client, err := net.NewControlClient(n.drand.opts.controlPort)
-			require.NoError(d.t, err)
+			if err != nil {
+				errDetector <- err
+				return
+			}
 			groupPacket, err := client.InitDKG(leaderNode.drand.priv.Public, nil, secret, d.beaconID)
-			require.NoError(d.t, err)
+			if err != nil {
+				errDetector <- err
+				return
+			}
 			group, err := key.GroupFromProto(groupPacket)
-			require.NoError(d.t, err)
+			if err != nil {
+				errDetector <- err
+				return
+			}
 
 			d.t.Logf("[RunDKG] NonLeader %s Finished. GroupHash %x", n.GetAddr(), group.Hash())
 
 			// We need to make sure the daemon is running before continuing
 			d.waitRunning(client, n)
-			wg.Done()
 		}(node)
 	}
 
+	select {
+	case err := <-errDetector:
+		required.NoError(err)
+	default:
+		fmt.Println("no activity")
+	}
+
 	// wait for all to return
 	wg.Wait()
```

Comment:
Good idea 👍🏻 

But I think you would need instead to `range` over the `errDetector` in order to catch all errors in the logs and also that needs to happen after the `wg.Wait()` otherwise the `default` clause will be triggered before the errors have a chance of being fired.

Also, please use `d.t.Log` instead of `fmt.Println`.

Reply:
looked more into this and you are correct. Thank you for pointing me to the right direction. 

that is noted but I don't think I will use that in this PR anymore. Thank you for the info